src/utils/globals.ts

```diff
@@ -254,54 +255,85 @@ export const calculateSecondsPerAttack = (
   return Math.round((baseAttackTime * 30) / (aspd / 100.0)) / 30;
 };
 
-export const getMaxPotStatIncrease = (characterObject: CharacterObject) => {
+// Returns an array of CharacterStatValues changes at each potential level,
+// with Pot2 at index 0 and Pot6 at index 4.
+export const getPotStatIncreases = (
+  characterObject: CharacterObject
+): PotentialStatChange[] => {
   const { potentialRanks } = characterObject;
 
-  const statChanges = {
-    health: 0,
-    attackPower: 0,
-    defense: 0,
-    artsResistance: 0,
-    attackSpeed: 0,
-    dpCost: 0,
-    redeployTimeInSeconds: 0,
-  };
+  const statChanges: PotentialStatChange[] = [];
 
   potentialRanks.forEach((pot) => {
     if (pot.buff === null) {
```

Comment:
I must have missed this earlier. We should probably use `== null` here (returns true if `undefined` or `null`, rather than `=== null` which returns false for `undefined`).

Reply:
Too used to non-Typescript so I just subconsciously use === for everything 🤣